src/Provider/AbstractProvider.php

```diff
@@ -408,11 +407,20 @@ protected function setResultUid(array &$result)
         }
     }
 
+    /**
+     * Generate a user object from a successful user details request.
+     *
+     * @param object $response
+     * @param AccessToken $token
+     * @return League\OAuth2\Client\Provider\UserInterface
+     */
+    abstract protected function prepareUserDetails(array $response, AccessToken $token);
```

Comment:
Made this protected abstract, since it really has no purpose as a public method.